src/tokens/ERC1155/presets/lootbox/ERC1155Lootbox.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import {IERC1155Lootbox, IERC1155LootboxFunctions} from "./IERC1155Lootbox.sol";
+import {ERC1155Items} from "@0xsequence/contracts-library/tokens/ERC1155/presets/items/ERC1155Items.sol";
+import {MerkleProof} from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import {IERC1155ItemsFunctions} from "@0xsequence/contracts-library/tokens/ERC1155/presets/items/IERC1155Items.sol";
+
+contract ERC1155Lootbox is ERC1155Items, IERC1155Lootbox {
+    bytes32 public merkleRoot;
+    uint256 public boxSupply;
+
+    mapping(address => uint256) private _commitments;
+    mapping(uint256 => bool) private _claimedIdxs;
+
+    constructor() ERC1155Items() {}
+
+    function setBoxContent(bytes32 _merkleRoot, uint256 _boxSupply) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        merkleRoot = _merkleRoot;
+        boxSupply = _boxSupply;
```

Comment:
Hmm. Do we also need to clear the `_claimedIdxs` and `_commitments` when this is updated? Or maybe we prevent updates when a commitment is pending? Or should we just accept that updating these may break existing commitments? Let me know your thoughts.

Reply:
As per slack convo, deciding to keep this as is with a usage warning in natspec.